include/xtypes/Assert.hpp

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2019, Proyectos y Sistemas de Mantenimiento SL (eProsima).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#ifndef EPROSIMA_XTYPES_ASSERT_HPP_
+#define EPROSIMA_XTYPES_ASSERT_HPP_
+
+#include <execinfo.h>
+#include <iostream>
+
+#if !defined(NDEBUG)
+
+#define xtypes_assert2_(cond, msg) xtypes_assert3_(cond, msg, false)
+
+#define xtypes_assert3_(cond, msg, bt)                                                                              \
+    {                                                                                                               \
+        if (!(cond))                                                                                                \
+        {                                                                                                           \
+            std::string msg__ = "[XTYPES]: ";                                                                       \
+            msg__.append(__FILE__ + std::string(":") + std::to_string(__LINE__) + std::string(" - "));              \
+            msg__.append(std::string("Assertion failed with message: "));                                           \
```

Comment:
Why not a stringstream?

Reply:
Do you see any advantage in this case?

Reply:
I think that stringstream works better with the string concatenation (but I am thinking now that could be equals). Anyway, the stringstream avoid you to use the std:string for each append.

Reply:
Once modified, it improves readability a lot. Nice!